document breaking change policy (#3262)
JuliaData/DataFrames.jl: docs/src/index.md

missing please kindly report an issue
[here](https://github.com/JuliaData/DataFrames.jl/issues/new)).

!!! note

    Breaking changes to public and documented API are avoided in
    DataFrames.jl where possible.

    The following changes are not considered breaking:

    * specific floating point values computed by operations may change at any
      time; users should rely only on approximate accuracy;

>     * in functions that use the default random number generator provided by the
>       Base Julia the specific random numbers computed may change at any time;

Reviewer: ```suggestion
    * in functions that use the default random number generator provided by
      Base Julia the specific random numbers computed may change across Julia versions;
```

    * if the changed functionality is classified as a bug;
    * if the changed behavior was not documented; two major cases are:
      1. in its implementation some function accepted a wider range of arguments
         that it was documented to handle - changes in handling of undocumented

>          arguments are not breaking;

Reviewer: ```suggestion
         arguments are not considered as breaking;
```

>       2. the type of returned value by the funtion changes, but it still follows
>          the contract specified in the documentation; for example if a function
>          is documented to return a vector then changing its type from `Vector`
>          to `PulledVector` is not breaking;

Reviewer: ```suggestion
      2. the type of the value returned by a function changes, but it still follows
         the contract specified in the documentation; for example if a function
         is documented to return a vector then changing its type from `Vector`
         to `PooledVector` is not considered as breaking;
```

    * error behavior: code that threw an exception can change exception type
      thrown or stop throwing an exception;
    * changes in display (how objects are printed);

>     * changes to the satate of global objets from Base Julia whose state

Reviewer 2: s/satate/state/

Reviewer 2: s/objets/objects/

      normally is considered volatile (e.g. state of global random number
      generator).

    All types and functions that are part of public API are guaranteed to go
    through a deprecation period before a breaking change is made to them

>     changed or they would be removed.

Reviewer 2: "changed"?

>     As a standard practice the change is implemented when a major release of
>     DataFrames.jl is made (e.g. functionalities deprecated in 1.x release
>     would be changed in 2.0 release).

Reviewer: ```suggestion
    The standard practice is that breaking changes are implemented when a major
    release of DataFrames.jl is made (e.g. functionalities deprecated in a 1.x release
    would be changed in the 2.0 release).
```

    In rare cases a breaking change might be introduced in a minor release.

>     In such case the changed behavior still goes through one minor release

Reviewer 2: "such a case" or "such cases"

>     during which it is deprecated. The situations where such breaking change

Reviewer 2: "such a breaking change" or "such breaking changes"

    might be allowed are (still such breaking changes will be avoided if
    possible):
    
    * the affected functionality was previously clearly identified in the

>       documentation as possible to be changed (for example in DataFrames.jl 1.4
>       release propagation rules of `:note`-style metadata are documented as
>       possible to be changed in the future);

Reviewer: ```suggestion
      documentation as being subject to changes (for example in DataFrames.jl 1.4
      release propagation rules of `:note`-style metadata are documented as such);
```
